src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs

```diff
@@ -283,6 +291,19 @@ public decimal Value
                 SetAndRaise(ValueProperty, ref _value, value);
             }
         }
+        
+        /// <summary>
+        /// Gets or sets the default value which is set if the <see cref="Value"/> is <see langword="null"/>.
+        /// </summary>
+        public decimal? DefaultValue
```

Comment:
This may be an unnecessary complication. The view that uses a `NumericUpDown` will have to decide what to do with a null value and then use an appropriate default. I wonder if such logic should be in the view itself.

That said, specifying a `DefaultValue` here can prevent null from ever occurring, correct? It also allows binding and is supported in XAML so there would be no code-behind required to avoid null values entirely.

Reply:
Hi,

We have done the same in MahApps back then becuase if your property in the ViewModel is not nullable, you will get an exception if the control sets null. I don't want to force the developer to use nullable properties everywhere. 

There are two options. Either set a flag `bool AllowsNull` or provide a default value. I like the default value approach because you can also set other default than 0. Consider you have a division operation with two values, the default may be 1 instead of 0, so you will not get a divide by zero exception.

One more thing to mention: Even if default value is set, the developer can still set the initial value to null. But the user cannot. 

I'll now wait for the feedback of the core team.

Happy coding
Tim

Reply:
I think what concerns me is this isn't a standardized property. Other controls (DatePicker) just provide a non-nullable property alternatively along with the nullable one.

I agree this is useful though and there are situations where non-nullabke makes things easier. Definitely looking forward to feedback from the core team.

Reply:
I think it would be better to follow conventions here. Have both a nullable and non-nullable property like DatePicker in WPF. Zero is an appropriate internal default but actually whatever is in the view model will be used. `DefaultValue` really does seem like an unnecessary complication.

For convention:
 1. It's best not to have a nullable property. However, the concept of no-user-selection needs to be supported. Some types allow this because the type itself can be null or represent an empty value (string, double.NaN, etc.).
 2. If there is no empty/null value provided by the data type, then use a nullable property like what you have. However, I think both a Nullable/Non-Nullable property should be supported. This would be like the CalendarDatePicker/DatePicker controls.

This means I wouldn't make it a breaking change to have the Value properties nullable. instead I would only add one new property for `SelectedValue` and track internally when the user has provided input. If no user-input, return null here. Otherwise, its the same as `Value` (which defaults to zero). I would double check what was done in the CalendarDatePicker code though.

Reply:
> This means I wouldn't make it a breaking change to have the Value properties nullable

Technically, it's already a breaking change to have a decimal, and we choose to go with it.
In existing code there won't be a source breaking changes in most of normal cases, like MVVM bindings. And in other places, where developers read value directly from the control in code behind, it should be easy to notice (compile time error) and fix it.

Reply:
Yea, I pushed for decimal type switch too. I like the idea of having both nullable and non-nullable properties to make this easiest for everyone. However, I absolutely would support making Value itself nullable too. As you said, this isn't that much more difficult to use. Some just don't want nullable ever and DatePicker handles things differently with two properties.

I would still get rid of DefaultValue though.